[skip changelog] Add documentation for upload components of the gRPC interface

per1234 opened the pull request:
**Please check if the PR fulfills these requirements**
- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls) before creating one)
- [x] The PR follows [our contributing guidelines](https://arduino.github.io/arduino-cli/CONTRIBUTING/#pull-requests)

* **What kind of change does this PR introduce?**
<!-- Bug fix, feature, docs update, ... -->
Add explanatory comments for the upload components of the gRPC interface to the source code. These comments will be included in the [generated gRPC interface documentation](https://arduino.github.io/arduino-cli/rpc/settings/).

* **What is the current behavior?**
<!-- You can also link to an open issue here -->
There is insufficient explanation of the gRPC interface in the documentation.

* **What is the new behavior?**
<!-- if this is a feature change -->
Improved documentation of the gRPC interface.

* **Does this PR introduce a breaking change?**
<!-- What changes might users need to make in their workflow or application due to this PR? -->
No

cmaglie commented:
We can also do the other way round: merge this one first and resolve the conflicts with #687 directly in #687.

per1234 commented:
I have now rebased on `master` and added documentation for the newly added `import_dir` field:
```protobuf
	// Custom path to a directory containing compiled files. When `import_dir` is
	// not specified, the standard build directory under `sketch_path` is used.
	string import_dir = 8;
```
